x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AbstractPhysicalOperationProviders.java

```diff
@@ -56,6 +60,23 @@ public abstract class AbstractPhysicalOperationProviders implements PhysicalOper
         this.analysisRegistry = analysisRegistry;
     }
 
+    private static Bucket findBucket(AggregateExec aggregateExec, NameId bucketId) {
+        Holder<Bucket> foundBucket = new Holder<>();
+        aggregateExec.forEachExpressionDown(NamedExpression.class, ne -> {
+            if (ne.id().equals(bucketId)) {
+                if (ne.children().size() > 0 && ne.children().get(0) instanceof Bucket bucket) {
+                    foundBucket.set(bucket);
+                    // TODO: Hack used when BUCKET is wrapped with ROUND. How to generalize it?
```

Comment:
Bucket technically can be used within any expression I think? Is this only for Round?